mlir/tools/mlir-tblgen/OpDefinitionsGen.cpp

```diff
@@ -2681,7 +2681,7 @@ void OpEmitter::genSeparateArgParamBuilder() {
                       {1}.regions, inferredReturnTypes)))
           {1}.addTypes(inferredReturnTypes);
         else
-          ::mlir::detail::reportFatalInferReturnTypesError({1});
+          ::mlir::detail::emitInferReturnTypesError({1});
```

Comment:
What happens post this in error case?

Reply:
It's up to the usage, similar to calling `op.verify`

Reply:
Correct, I'm following the same pattern used in `<Type>::getChecked` and `<Type>::parse` for instance, which afaict expect you to check whether a diagnostic was emitted to signal an error state. I don't think this is a great pattern necessarily, we can do something else, but it's the pattern that seems to (1) have prior use in the codebase and (2) appears to locally best fit the needs.

Reply:
Actually no: `getChecked` returns a `null` Attribute (or Type) which you can check. This is not the case here, the MLIR builder API has no provision to fail building an operation, and your patch does not do this right now.

Reply:
I'm binding the `create` methods, see a concrete example [here](https://github.com/llvm/llvm-project/pull/165488#issuecomment-3518378198). The generic builders are not very ergonomic, and we write C++ create methods for many ops with more complex construction requirements. I want to use those rather than re-implementing them on the Python side.